### PR TITLE
Nfdiv 2622 fix caseworker offline doc notifications

### PR DESCRIPTION
**Pull request by FaisalMoJ**
### Change description ###

Currently the functionality related to sending notifications is in about to submit call back. Sometimes, the caseworker-offline-document-verified event does not complete successfully and with that notifications are sent but case does not progress to the next state. This change moves the notifications to the submitted callback and sends notifications only when event completes successfully.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2622

**Before merging a pull request make sure that:**

- [X] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

### CODE REVIEW

**Review by tomxelliott — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerOfflineDocumentVerifiedFT.java [hunk 1]
```diff
@@ -28,7 +29,7 @@ public class CaseworkerOfflineDocumentVerifiedFT extends FunctionalTestSuite {
         "classpath:responses/response-caseworker-offline-document-verified-d84-about-to-submit.json";
 
     @Test
-    public void shouldSendAppliedForCoLetterIfD84Selected() throws IOException {
+    public void shouldGenerateReclassifiedScannedDocumentsIfD84Selected() throws IOException {
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerOfflineDocumentVerifiedFT.java:32]: shouldReclassifyScannedDocumentsIfD84Selected ?

File: src/functionalTest/resources/responses/response-caseworker-offline-document-verified-d84-about-to-submit.json [hunk 2]
```diff
@@ -20,28 +20,6 @@
...
-      "value": {
-        "documentLink": {
-          "document_url": "${json-unit.any-string}",
-          "document_filename": "${json-unit.any-string}",
-          "document_binary_url": "${json-unit.any-string}"
-        },
-        "documentFileName": "${json-unit.any-string}",
-        "documentType": "appliedForCoLetter"
-      }
-    }, {
```
Comment (nitinprabhu) [inline src/functionalTest/resources/responses/response-caseworker-offline-document-verified-d84-about-to-submit.json:44]: was this duplicate ?

**Review by nitinprabhu — APPROVED** [review]

File: src/functionalTest/resources/responses/response-caseworker-offline-document-verified-d84-about-to-submit.json (hunk 2 above)
Comment (FaisalMoJ) [reply src/functionalTest/resources/responses/response-caseworker-offline-document-verified-d84-about-to-submit.json:44]: @nitinprabhu These documents are generated during notification. That is why i removed it from the response when i moved the notification related code to submitted callback

File: src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerOfflineDocumentVerifiedFT.java (hunk 1 above)
Comment (FaisalMoJ) [reply src/functionalTest/java/uk/gov/hmcts/divorce/caseworker/CaseworkerOfflineDocumentVerifiedFT.java]: Renamed unit test as suggested by Nitin

**Pull request merged**